ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,671 @@
+from functools import cached_property
+from hashlib import sha1
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+from ansible_collections.arista.avd.plugins.filter.snmp_hash import hash_passphrase
+
+
+class AVDStructConfig(AvdFacts):
+
+    @cached_property
+    def _mgmt_interface_vrf(self):
+        """
+        Returns the value for mgmt_interface_vrf variable used in static_routes, name_Server,
+        vrfs and management_interfaces data-models
+        """
+        return get(self._hostvars, 'mgmt_interface_vrf')
+
+    @cached_property
+    def _mgmt_gateway(self):
+        """
+        Returns the value for mgmt_gateway variable used in static_routes and management_interfaces data-models
+        """
+        return get(self._hostvars, 'mgmt_gateway')
+
+    @cached_property
+    def _switch(self):
+        """
+        Returns avd_switch_facts
+        """
+        return get(self._hostvars, 'switch', required=True)
+
+    @cached_property
+    def _platform_settings(self):
+        """
+        Returns the value for switch.platform_settings fact used in queue_monitor_length, tcam_profile, platform
+        and eos_cli data-models
+        """
+        return get(self._switch, 'platform_settings')
+
+    @cached_property
+    def _mgmt_ip(self):
+        """
+        Returns the value for switch.mgmt_ip fact used in snmp_server and management_interfaces data-models
+        """
+        return get(self._switch, 'mgmt_ip')
+
+    @cached_property
+    def _hostname(self):
+        """
+        hostname variable set based on switch.hostname fact
+        """
+        return get(self._switch, "hostname", required=True)
```

Comment:
everywhere where we pull a *required* variable from a local _switch or similar, we need to have the full key name in `org_key`. So in this example add `org_key="switch.hostname"`. This is required to produce useful error messages